### ISSUE TITLE
Generate built-in `Range`

### ISSUE THREAD

**Issue by stijnmoreels** [fscheck/FsCheck#621]
Hi! 👋 
Was thinking of adding a built-in generator of the `System.Range` struct, as this would expand the default generator functionality.
Am currently using this:
```fsharp
let range = gen {
  let! min = Gen.choose (0, Int32.MaxValue)
  let! max = Gen.choose (min, Int32.MaxValue)
  return Range (min, max) }
```
To make sure that the minimum is always lower than the maximum. But, with some additional options, one could include `Range.StartAt` and `Range.EndAt` too.

Just a suggestion when using the awesome FsCheck library 😉 .

**Comment by kurtschelfthout** [comment]
Related: there's also [Interval](https://github.com/fscheck/FsCheck/blob/master/src/FsCheck/Arbitrary.fs#L78)

**Comment by stijnmoreels** [comment]
Ah, shoot, didn't see that one. Maybe that would be ok, too.
I'll try that. Thx!

**Comment by stijnmoreels** [comment]
Aha, found everything I needed. Thx for this! 👍

**Comment by stijnmoreels** [comment]
Maybe an idea to create an extra generator that generate the built-in `System.Range` from an `Interval`?
But that may be overdoing it.

**Comment by kurtschelfthout** [comment]
> Maybe an idea to create an extra generator that generate the built-in System.Range from an Interval?

Yeah I think that makes sense. Or even extract the code and parametrize the contructor, could be slightly faster without creating an intermediate type.

For standard types like `System.Range` I think it certainly makes sense to have them built-in.

**Comment by stijnmoreels** [comment]
Great! Happy to pick this up.
